src/netaddress.h

```diff
@@ -533,6 +533,24 @@ class CSubNet
     friend bool operator==(const CSubNet& a, const CSubNet& b);
     friend bool operator!=(const CSubNet& a, const CSubNet& b) { return !(a == b); }
     friend bool operator<(const CSubNet& a, const CSubNet& b);
+
+    SERIALIZE_METHODS(CSubNet, obj)
+    {
+        READWRITE(obj.network);
+        if (obj.network.IsIPv4()) {
+            // Before commit 102867c587f5f7954232fb8ed8e85cda78bb4d32, CSubNet used the last 4 bytes of netmask
+            // to store the relevant bytes for an IPv4 mask. For compatibility reasons, keep doing so in
+            // serialized form.
+            unsigned char dummy[12] = {0};
+            READWRITE(dummy);
+            READWRITE(MakeSpan(obj.netmask).first(4));
+        } else {
+            READWRITE(obj.netmask);
+        }
+        READWRITE(obj.valid);
+        // Mark invalid if the result doesn't pass sanity checking.
+        SER_READ(obj, if (obj.valid) obj.valid = obj.SanityCheck());
```

Comment:
When you are re-introducing this, it might be good to cleanup the complexity in this method or maybe just use the existing json serialization?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/10102#discussion_r690922929

> When you are re-introducing this, it might be good to cleanup the complexity in this method or maybe just use the existing json serialization?

Thanks, exposed json functions in #22848, and now using it in this PR